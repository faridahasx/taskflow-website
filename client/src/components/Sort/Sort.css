#sort-ul {
    width: 230px;
    max-height: 300px;
    overflow: auto;
}

.sf-wrapper.sort-dialog {
    position: absolute;
    left: -5px;
    top: -5px;
    z-index: 100;
}

.sort-component,
#sort-ul button {
    width: 100%;
}

#sort-ul button:hover,
#sort-ul .sort-field-active {
    color: var(--main-green);
}

.sort-field-icon {
    width: 40px;
    height: 40px;
    margin-right: 10px;
}

@media all and (max-width: 376px) {
    .sf-wrapper.sort-dialog {
        width: calc(100vw - 40px);
        left: -100px;
    }

    #sort-ul {
        width: 100%;
    }
}