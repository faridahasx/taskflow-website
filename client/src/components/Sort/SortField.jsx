import PropTypes from "prop-types";
// MUI components
import CheckIcon from "@mui/icons-material/Check";

const SortField = (props) => {
  const { name, value, currentSort, handleSorting } = props;
  const handleClick = () => handleSorting(value);

  return (
    <li className="center sort-component">
      <button
        className={`sort-field flex ${
          currentSort === value ? " sort-field-active" : ""
        }`}
        title={name}
        onClick={handleClick}
      >
        <span className="center sort-field-icon">
          <CheckIcon />
        </span>
        {name}
      </button>
    </li>
  );
};

SortField.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  currentSort: PropTypes.string,
  handleSorting: PropTypes.func.isRequired,
};

export default SortField;
