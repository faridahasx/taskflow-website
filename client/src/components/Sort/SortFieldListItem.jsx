import PropTypes from "prop-types";
// MUI components
import CheckIcon from "@mui/icons-material/Check";

const SortFieldListItem = (props) => {
  const { handleSorting, sortValue, value, name } = props;
  const handleClick = () => handleSorting(value);

  return (
    <li className="center sf-component">
      <button
        className={`sort-field flex ${
          sortValue === value ? " sort-field-active" : ""
        }`}
        title={name}
        onClick={handleClick}
      >
        <span className="center sort-field-icon">
          <CheckIcon />
        </span>
        {name}
      </button>
    </li>
  );
};

SortFieldListItem.propTypes = {
  handleSorting: PropTypes.func.isRequired,
  sortValue: PropTypes.string,
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default SortFieldListItem;
