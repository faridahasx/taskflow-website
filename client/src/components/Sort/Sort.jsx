import PropTypes from "prop-types";
// MUI components
import SwapVertIcon from "@mui/icons-material/SwapVert";
// Constant data
import sortFields from "../../constants/sortFields";
// Custom hooks
import useClickOutside from "../../hooks/useClickOutside";
// Components
import Dialog from "../SortAndFiltersWrapper/SortAndFiltersWrapper";
import SortFieldListItem from "./SortFieldListItem";
// Styles
import "./Sort.css";

const Sort = (props) => {
  // Destructuring props
  const { handleClose, handleSorting, sortValue } = props;
  // Ref for handling clicks outside the container
  const containerRef = useClickOutside(handleClose);

  // Render compomnent
  return (
    <Dialog
      handleClose={handleClose}
      heading={"Sort By"}
      Icon={<SwapVertIcon />}
      className="sort-dialog"
      containerRef={containerRef}
    >
      <ul id="sort-ul" data-testid="sort">
        {sortFields.map((s) => (
          <SortFieldListItem
            handleSorting={handleSorting}
            sortValue={sortValue}
            value={s.value}
            name={s.name}
            key={s.value}
          />
        ))}
      </ul>
    </Dialog>
  );
};

Sort.propTypes = {
  handleClose: PropTypes.func.isRequired,
  handleSorting: PropTypes.func.isRequired,
  sortValue: PropTypes.string,
};

export default Sort;
