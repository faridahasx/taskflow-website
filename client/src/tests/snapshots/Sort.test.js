import { render } from "@testing-library/react";
import Sort from "../../components/Sort/Sort";

const mockFn = jest.fn();

test("Sort snapshot", () => {
  const { asFragment } = render(
    <Sort handleClose={mockFn} handleSorting={mockFn} sortValue={""} />
  );
  expect(asFragment()).toMatchSnapshot("Sort");
});
