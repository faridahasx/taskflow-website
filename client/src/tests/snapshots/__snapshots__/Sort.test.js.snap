// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort snapshot: Sort 1`] = `
<DocumentFragment>
  <div
    class="sf-wrapper sort-dialog"
  >
    <div
      class="sf-top flex"
    >
      <button
        class="icon "
        title="Close"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
      <h3
        class="heading-text flex"
      >
        Sort By
      </h3>
    </div>
    <ul
      data-testid="sort"
      id="sort-ul"
    >
      <li
        class="center sf-component"
      >
        <button
          class="sort-field flex "
          title="First start first"
        >
          <span
            class="center sort-field-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </span>
          First start first
        </button>
      </li>
      <li
        class="center sf-component"
      >
        <button
          class="sort-field flex "
          title="Last start first"
        >
          <span
            class="center sort-field-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </span>
          Last start first
        </button>
      </li>
      <li
        class="center sf-component"
      >
        <button
          class="sort-field flex "
          title="First finish first"
        >
          <span
            class="center sort-field-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </span>
          First finish first
        </button>
      </li>
      <li
        class="center sf-component"
      >
        <button
          class="sort-field flex "
          title="Last finish first"
        >
          <span
            class="center sort-field-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </span>
          Last finish first
        </button>
      </li>
      <li
        class="center sf-component"
      >
        <button
          class="sort-field flex "
          title="First update first"
        >
          <span
            class="center sort-field-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </span>
          First update first
        </button>
      </li>
      <li
        class="center sf-component"
      >
        <button
          class="sort-field flex "
          title="Last update first"
        >
          <span
            class="center sort-field-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </span>
          Last update first
        </button>
      </li>
      <li
        class="center sf-component"
      >
        <button
          class="sort-field flex "
          title="First created first"
        >
          <span
            class="center sort-field-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </span>
          First created first
        </button>
      </li>
      <li
        class="center sf-component"
      >
        <button
          class="sort-field flex "
          title="Last created first"
        >
          <span
            class="center sort-field-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </span>
          Last created first
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
